backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/RunVmCommandBase.java

```diff
@@ -146,6 +150,7 @@ protected void cleanupPassthroughVnics(Guid hostId) {
     @Override
     public void rerun() {
         decreasePendingVm();
+        getVdsManager().unpinVmCpus(getVmId());
```

Comment:
we can't move it.
decreasePendingVm() happens also when running succeed. we don't want to clear these resources in such case.

Reply:
> regarding rerun() - i added it since we execute the flow all over again. that suppose to include the scheduling including the allocation. so it's ok to release it here.

Sure, my point was that it was missing also for rerun of RunVm so this way we kill two birds in one shot

> about moving it to decreasePendingVm() - we can't move it. decreasePendingVm() happens also when running succeed. we don't want to clear these resources in such case.

Yeah, that would be a massive change to do now but I'd imagine we could better align the code with other pending resources by adding the pinning not as pending when the VM powers on. But yes, with the current design we can't easily do that..